# `texts` as Texts

In [1]:
# IMPORTS
import json
import nltk

import random
# import matplotlib.pyplot as plt

In [47]:
# DATA
with open("texts.json", 'r') as f:
    texts = json.load(f)

# Check to see that the data loaded as expected
texts[30:35]

[' Anyway... maybe I shouldn’t be so serious with you. Maybe I should just ask you about your day and make you smile. I do tend to get a bit deep, though — that is my curse lol. I’m very playful as well, however 🙈\n ',
 ' Idk-  I don’t really try to meet people.  I just work, come home, sleep &amp;amp; then do it again the next day.  Maybe that’s some of the problem.  Who knows... I’m starting to just not gaf anymore.  Maybe a race war will kick off &amp;amp; I’ll have purpose then.  Fug it\n ',
 ' I am nice.  I’m a good person.  I mean, I’m annoying at times I’m sure &amp;amp; I can get bitchy like any woman does-  but I’m not a cunt &amp;amp; I’m pretty chill.  I’ve never been with a non white.  I’ve only been with 2 people in my life... both were white.  I don’t have herpes lmao!    I think I’m pretty average as far as looks go.  I’m not some Swedish super model, but I’m not fugly.  I’m in shape.  I don’t have a nigger ass.   Idk-  I’m just a normal average lass kicking off my 30’s 

In [49]:
from lxml import etree

In [50]:
def remove_html_tags(text):
    parser = etree.HTMLParser()
    tree = etree.fromstring(text, parser)
    return etree.tostring(tree, encoding='unicode', method='text')

In [52]:
threefive = remove_html_tags(texts[35])

In [53]:
plaintexts = [remove_html_tags(i) for i in texts]

TypeError: Type 'NoneType' cannot be serialized.

In [3]:
# Tokenize all the texts
words = [ nltk.tokenize.word_tokenize(text.lower()) for text in texts ]

# And check to see that went according to plan
print(words[20:22])

[['are', 'you', 'working', 'today', ',', 'beautiful', 'girl', ',', 'or', 'are', 'you', 'hanging', 'with', 'fam', '?'], ['i', '’', 'm', 'working', '.', 'the', 'only', 'days', 'i', 'have', 'off', 'for', 'the', 'next', '2', 'weeks', 'is', 'christmas', 'day', '&', 'amp', ';', 'amp', ';', 'new', 'year', '’', 's', 'day', '.', ':', '(', '.', 'i', 'need', 'to', 'get', 'another', 'job', '.', 'this', 'one', 'is', 'killing', 'me']]


With all our texts now rendered as a lists of tokens, let's take a look at how long the texts are. We're first just going to get a list of the token counts, then we will determine the texts that fall within number ranges that are generated by some initial hand inspection. My logic is, for example, that texts that are less than 20 words are not going to offer much in the way of narrative, and, as it turns out, the longer texts are stuffed with HTML.

In [4]:
# Count the tokens in each text
counts = [ len(i) for i in words ]

# Collect the number of texts within a given length range
print(sum([i < 20 for i in counts]))
print(sum([20 < i < 50 for i in counts]))
print(sum([50 < i < 100 for i in counts]))
print(sum([100 < i < 200 for i in counts]))
print(sum([200 < i < 500 for i in counts]))
print(sum([500 < i for i in counts]))

47344
13819
6212
1408
394
16


In [5]:
# Let's add those totals up
total_counts = 47344 + 13819 + 6212 + 1408 + 394 + 16
print(total_counts)

# And compare them against the total number of texts
print(len(texts))

# These two totals turn up a discrepancy (explained below)
print( len(texts) - total_counts)

69193
70596
1403


*Sigh*. 1403 texts are missing! For now, I am not going to worry about it. I have enough texts with which to work.

In [6]:
word_0 =  [ i for i in words if len(i) <  20 ]
word_20 = [ i for i in words if 20 < len(i) <  50 ]
word_50 = [ i for i in words if 50 < len(i) < 100 ]
word100 = [ i for i in words if 100 < len(i) < 200 ]
word200 = [ i for i in words if 200 < len(i) < 500 ]
word500 = [ i for i in words if len(i) > 500 ]

In [9]:
workable_lengths = len(word_20) + len(word_50) + len(word100) + len(word200)
print(workable_lengths)

21833


In [10]:
workable = [ i for i in words if 20 < len(i) < 500 ]

In [19]:
len(workable)

22326

In [24]:
len(workable[8])

70

In [71]:
for i in random.sample(workable, 3):
    print(" ".join(i))

you 're piquing my interest . i want to know what podcast you 're talking about . you do n't have to tell me , but i wan na know .
matk schlieder @ matkschlieder · 8s replying to @ leoncosewgirl < a href= '' https : //gab.com/laylaalisha11 '' class= '' u-url mention '' rel= '' nofollow noopener noreferrer '' target= '' _blank '' > @ laylaalisha11 < /a > and 2 others but the lockdowns are not and never have been about the virus , they are about control and obeying the deep state psyop and the demands of the chicoms so as to take over the country and our economy . and it ’ s still continuing due to the hysteria and cowardice of the people .
hey dan . you do great work . i am migrating off twitter . just followed you on parler as well .


In [39]:
import pickle

with open("workable.pkl", "wb") as fp:
    pickle.dump(workable, fp)

In [38]:
import json

with open("workable.json", 'w') as f:
    json.dump(workable, f, indent=2) 